# Binomial Distribution

In [1]:
from scipy.stats import binom

In [2]:
n = 10
p = 0.9

In [4]:
probabilities = []

for k in range(n + 1):
    probability = binom.pmf(k, n, p)
    probabilities.append(probability)

    print(f"{k} - {probability}")

0 - 9.999999999999978e-11
1 - 8.999999999999978e-09
2 - 3.6449999999999943e-07
3 - 8.747999999999991e-06
4 - 0.00013778099999999974
5 - 0.0014880347999999982
6 - 0.011160260999999996
7 - 0.05739562799999997
8 - 0.1937102444999998
9 - 0.38742048899999976
10 - 0.3486784401000001


The probability to get eight or less success if the underlaying success rate is 90%:

In [7]:
prop_8orless = sum(probabilities[:9])
print(prop_8orless * 100, "%")

26.390107089999976 %


# Beta Distribution
Beta Distribution can be used to see the likelihood of different underlying probabilities for an event to occur given *a* succcesses and *b* failures.

In [8]:
from scipy.stats import beta

We have observed 8 success and 2 failures.

In [10]:
a = 8
b = 2

Now we can calculate the probability that the underlying probability of success is 90% or less

In [11]:
p = beta.cdf(0.9, a, b)
p

0.7748409780000002

So we have a 77.48% probability that the underlying probability distribution is 90% or less.

Next, let's calculate the probability that the underlying success rate is over 90%, given we have 8 success and 2 failures.

In [12]:
p_ = 1 - p
p_

0.22515902199999982

So the probability, that we have a higher probability distribution than 90%, is 22.5%

## Second test
A second test with 36 iterations has shown, that we have 30 successes and 6 failures.

In [13]:
a = 30
b = 6

In [14]:
p = beta.cdf(0.9, a, b)
p

0.868364225158163

In [15]:
1 - p

0.13163577484183697

We see, that it is now less likely, that the underlying probability rate is equal or above 90%.

## What is the probability that we have a underlying rate between 80% and 90%?
Given 8 successes and 2 failures

In [16]:
a = 8
b = 2

In [17]:
p = beta.cdf(0.9, a, b) - beta.cdf(0.8, a, b)
p

0.33863336199999994

The probability in that case is 33.86%

# Exercises

## Exercise 1
30% chance of rain today, and a 40% chace your umbrella order will arrive in time. What is the probability it will rain and your umbrella will arrive?

In [19]:
p = 0.3 * 0.4
p

0.12

## Exercise 2
What is the prob it will not rain or your umbrella arrives?

In [22]:
p = (1-0.3) + 0.4 - (0.3 * 0.4)
p

0.9800000000000001

## Exercise 3
If it rains there is a 20% change your umbrella will arrive on time.
What is the prob it will rain and your umbrella will arrive on time?

In [23]:
p = 0.3 * 0.2
p

0.06

## Exercise 4
You have 137 passengers booked on a flight from Las Vegas to Dallas. However, it is las vegas on a sunday morning and you estimate each passenger is 40% likely to not show up.
How likely is it at least 50 passengers will not show up?

In [32]:
p = 0.4

likely = 0
for i in range(50, 138):
    likely += binom.pmf(i, 137, p)
print(f"{likely * 100}")

82.20955881474255


## Exercise 5
You flipped a coin 19 times and got heads 15 times and tails 4 times.
Do you think this coin has any good propability of being fair?

In [35]:
a = 15
b = 4

# What is the probability, that we have an underlying probability rate between 0.45 and 0.55, which I would count as fair
p = beta.cdf(0.55, a, b) - beta.cdf(0.45, a, b)
p

0.010987919688451336

I would say, that this coin is not fair.